interfaces/Portal/src/app/pages/home/components/programs-list/programs-list.component.ts

```diff
@@ -27,10 +27,15 @@ export class ProgramsListComponent implements OnInit {
 
   private async refresh() {
     this.loading = true;
-    const programs = await this.programsService.getAllPrograms();
+    const programIds = await this.programsService.getAllProgramIds();
     this.programStats = await this.programsService.getAllProgramsStats(
-      programs.map((p) => p.id),
+      programIds.map((id) => id),
     );
+    const programs = [];
+    for (const programId of programIds) {
+      const program = await this.programsService.getProgramById(programId);
+      programs.push(program);
+    }
```

Comment:
TS likes it when you use `Array.map`, because then the resulting `programs` will have the correct type.
```suggestion
    const programs = Promise.all(programIds.map(async (programId) =>
      await this.programsService.getProgramById(programId);
    ));
```

(Check the syntax because I'm not 100% sure about it)

Reply:
Nice suggestion 👍